pydicom/tests/test_dataset.py

```diff
@@ -313,6 +313,17 @@ def testUpdate(self):
             'Johnny',
             "set by tag failed")
 
+    def test_dir_subclass(self):
+        """Dataset.__dir__ returns class specific dir"""
+        class DSP(Dataset):
+            def test_func(self):
+                pass
+
```

Comment:
Could you add an assert to check that you have an attribute/method of `Dataset`.